olps/datasources/correlation_driven_datasource.py

```diff
@@ -0,0 +1,63 @@
+from .datasource import DataSource
+import numpy as np
+import math
+
+
+class CorrelationDrivenDataSource(DataSource):
+
+    window = 2
+    rho = 0.2 
+    similarity_set = None
+    similarity_set_size = 0
+    # modifying the constructors to include the window size
+    def __init__(self, initial_prices: np.array, window):
```

Comment:
This could be slightly improved by making some of these options settable in the constructor with default values:

```python
window: int
rho: float
similarity_set: np.array
similarity_set_size: int

def __init__(self, initial_prices: np.array, window=2, rho=2):
    # other code
    self.window = window
    self.similarity_set = None
    self.similarity_set_size = 0
    self.rho = rho
```

Why do this? It keeps the type definitions at the top and makes the defaults explicit (as well as configurable).